tests/test_invenio_communities.py

```diff
@@ -145,3 +146,22 @@ def test_model_featured_community(app):
             start_date=t1 + timedelta(days=2)) is comm1
         assert FeaturedCommunity.get_featured_or_none(
             start_date=t1 + timedelta(days=4)) is comm2
+
+
+def test_oaipmh_sets(app):
+    """Test the OAI-PMH Sets creation."""
+    with app.app_context():
+        # Create a user
+        user1 = create_test_user()
+        comm1 = Community(id='comm1',
+                          id_user=user1.id,
+                          title='Title1',
+                          description='Description1')
+        db.session.add(comm1)
+        db.session.commit()
```

Comment:
Thank you for the informative reference but that wasn't where my confusion came from ;-)

I wrongly assumed that "after_insert" signal is sent after the ``session.add``, rather than SQL INSERT, thus I expected receiver function to be called without the need for session to commit.